pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -227,38 +355,50 @@ func buildRegistrationEntriesCache(ctx context.Context, ds datastore.DataStore,
 
 		entries, err := api.RegistrationEntriesToProto(resp.Entries)
 		if err != nil {
-			return 0, fmt.Errorf("failed to convert registration entries: %w", err)
+			return 0, nil, fmt.Errorf("failed to convert registration entries: %w", err)
 		}
 
 		for _, entry := range entries {
 			cache.UpdateEntry(entry)
 		}
 	}
 
-	return lastEventID, nil
+	return lastEventID, missedRegistrationEntryEvents, nil
 }
 
 // buildAttestedNodesCache Fetches all attested nodes and adds the unexpired ones to the cache
-func buildAttestedNodesCache(ctx context.Context, ds datastore.DataStore, clk clock.Clock, cache *authorizedentries.Cache) (uint, error) {
-	lastEventID, err := ds.GetLatestAttestedNodeEventID(ctx)
-	if err != nil && status.Code(err) != codes.NotFound {
-		return 0, fmt.Errorf("failed to get latest attested node event id: %w", err)
+func buildAttestedNodesCache(ctx context.Context, ds datastore.DataStore, clk clock.Clock, cache *authorizedentries.Cache) (uint, map[uint]time.Time, error) {
+	// Gather any events that may have been skipped during restart
+	var lastEventID uint
+	missedAttestedNodeEvents := make(map[uint]time.Time)
```

Comment:
We need to look back in the event history to see if there is any gaps because the event might commit after we get all entries.

Reply:
@MarcosDY  Yes, it is required.  When a long running write transaction that is modify SPIRE Entries doesn't commit because it is hung, it leaves gaps in the auto increment keys of the event table.  By tracking which event ids in the event table are missing, we can then go back and explicitly poll the skipped event ids prior to scanning the table for new entry event ids.  This is the best that can be done at the moment to ensure that we don't skip event ids when other transactions push the last seen event id beyond those that are stuck in hung database transactions.